pennylane/spin/spin_hamiltonian.py

```diff
@@ -94,3 +97,184 @@ def transverse_ising(
+                      ``(num_spins, num_spins)``. Default value is [1.0].
+       interaction (float or List[float]): Coulomb interaction between spins, it can be a constant or a list of length ``num_spins``.
+       boundary_condition (bool or list[bool]): Defines boundary conditions different lattice axes,
+           default is ``False`` indicating open boundary condition.
+       neighbour_order (int): Specifies the interaction level for neighbors within the lattice.
+                       Default is 1 (nearest neighbour).
+       mapping (str): Specifies the fermion-to-qubit mapping. Input values can be
+                      ``'jordan_wigner'``, ``'parity'`` or ``'bravyi_kitaev'``.
+
+    Returns:
+       pennylane.operator: Hamiltonian for the Fermi-Hubbard model.
+
+    **Example**
+
+    >>> n_cells = [2,2]
+    >>> h = [[0.5]]
+    >>> u = [[1.0]]
+    >>> spin_ham = fermihubbard("Square", n_cells, hopping=h, interaction=u)
+    >>> print(spin_ham)
+
+    """
+
+    lattice = _generate_lattice(lattice, n_cells, boundary_condition, neighbour_order)
+
+    if hopping is None:
+        hopping = [1.0]
+    elif isinstance(hopping, (int, float, complex)):
+        hopping = [hopping]
+
+    hopping = math.asarray(hopping)
+
+    if hopping.shape not in [(neighbour_order,), (lattice.n_sites, lattice.n_sites)]:
+        raise ValueError(
+            f"hopping should be a number or an array of shape {neighbour_order}x1 or {lattice.n_sites}x{lattice.n_sites}"
+        )
+
+    spin = 2
+    hopping_ham = 0.0
+    if hopping.shape == (neighbour_order,):
+        for edge in lattice.edges:
+            for s in range(spin):
+                i, j, order = edge[0], edge[1], edge[2]
+                s1 = i * spin + s
+                s2 = j * spin + s
+                hopping_term = -hopping[order] * (
+                    FermiWord({(0, s1): "+", (1, s2): "-"})
+                    + FermiWord({(0, s2): "+", (1, s1): "-"})
+                )
+                hopping_ham += hopping_term
+    else:
+        for edge in lattice.edges:
+            for s in range(spin):
+                i, j, order = edge[0], edge[1], edge[2]
+                s1 = i * spin + s
+                s2 = j * spin + s
+                hopping_term = -hopping[i][j] * (
+                    FermiWord({(0, s1): "+", (1, s2): "-"})
+                    + FermiWord({(0, s2): "+", (1, s1): "-"})
+                )
+                hopping_ham += hopping_term
+
+    int_term = 0
+    if isinstance(interaction, (int, float, complex)):
+        interaction = math.ones(lattice.n_sites) * interaction
+
+    for i in range(lattice.n_sites):
+        up_spin = i * spin
+        down_spin = i * spin + 1
+        int_term += interaction[i] * FermiWord(
+            {(0, up_spin): "+", (1, up_spin): "-", (2, down_spin): "+", (3, down_spin): "-"}
+        )
+
+    hamiltonian = hopping_ham + int_term
+
+    if mapping not in ["jordan_wigner", "parity", "bravyi_kitaev"]:
```

Comment:
Will qubit_observable function not return a similar error? If yes, do we need to duplicate error here? 🤔

Reply:
qubit_observable function doesn't return the error, the error is based in the operator functions in qchem. Something to be considered later is that we move it to qubit_observable function for all other functions.